lib/tasks/mastodon.rake

```diff
@@ -4,6 +4,258 @@ require 'optparse'
+            q.required true
+            q.default "files.#{env['LOCAL_DOMAIN']}"
+            q.modify :strip
+          end
+
+          env['AWS_ACCESS_KEY_ID'] = prompt.ask('Wasabi access key:') do |q|
+            q.required true
+            q.modify :strip
+          end
+
+          env['AWS_SECRET_ACCESS_KEY'] = prompt.ask('Wasabi secret key:') do |q|
+            q.required true
+            q.modify :strip
+          end
+        when 'Minio'
+          env['S3_ENABLED']  = 'true'
+          env['S3_PROTOCOL'] = 'https'
+          env['S3_REGION']   = 'us-east-1'
+
+          env['S3_ENDPOINT'] = prompt.ask('Minio endpoint URL:') do |q|
+            q.required true
+            q.modify :strip
+          end
+
+          env['S3_PROTOCOL'] = env['S3_ENDPOINT'].start_with?('https') ? 'https' : 'http'
+          env['S3_HOSTNAME'] = env['S3_ENDPOINT'].gsub(/\Ahttps?:\/\//, '')
+
+          env['S3_BUCKET'] = prompt.ask('Minio bucket name:') do |q|
+            q.required true
+            q.default "files.#{env['LOCAL_DOMAIN']}"
+            q.modify :strip
+          end
+
+          env['AWS_ACCESS_KEY_ID'] = prompt.ask('Minio access key:') do |q|
+            q.required true
+            q.modify :strip
+          end
+
+          env['AWS_SECRET_ACCESS_KEY'] = prompt.ask('Minio secret key:') do |q|
+            q.required true
+            q.modify :strip
+          end
+        end
+
+        if prompt.yes?('Do you want to access the uploaded files from your own domain?')
+          env['S3_CLOUDFRONT_HOST'] = prompt.ask('Domain for uploaded files:') do |q|
+            q.required true
+            q.default "files.#{env['LOCAL_DOMAIN']}"
+            q.modify :strip
+          end
+        end
+      end
+
+      prompt.say "\n"
+
+      env['SMTP_SERVER'] = prompt.ask('SMTP server:') do |q|
+        q.required true
+        q.default 'smtp.mailgun.org'
+        q.modify :strip
+      end
+
+      env['SMTP_PORT'] = prompt.ask('SMTP port:') do |q|
+        q.required true
+        q.default 587
+        q.convert :int
+      end
+
+      env['SMTP_LOGIN'] = prompt.ask('SMTP username:') do |q|
+        q.required true
+        q.modify :strip
+      end
+
+      env['SMTP_PASSWORD'] = prompt.ask('SMTP password:') do |q|
+        q.required true
+        q.echo false
+      end
+
+      env['SMTP_FROM_ADDRESS'] = prompt.ask('E-mail address to send e-mails "from":') do |q|
+        q.required true
+        q.default "Mastodon <notifications@#{env['LOCAL_DOMAIN']}>"
+        q.modify :strip
+      end
+
+      prompt.say "\n"
+      prompt.say 'This configuration will be written to .env.production'
+
+      if prompt.yes?('Save configuration?')
+        cmd = TTY::Command.new(printer: :quiet)
+
+        File.write(Rails.root.join('.env.production'), "# Generated with mastodon:setup on #{Time.now.utc}\n\n" + env.each_pair.map { |key, value| "#{key}=#{value}" }.join("\n") + "\n")
```

Comment:
This does not include comments, and I think it puts keys in a random order. This would not make it easy to edit the file afterward.

Reply:
That doesn't really matter. Nobody canceled `.env.production.sample`. Order of the params does not matter at all for the machine.